packages/pigeon/example/app/linux/messages.g.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+// Autogenerated from Pigeon (v11.0.1), do not edit directly.
+// See also: https://pub.dev/packages/pigeon
+
+#ifndef PIGEON_MESSAGES_G_H_
+#define PIGEON_MESSAGES_G_H_
+
+#include <flutter_linux/flutter_linux.h>
+
+G_BEGIN_DECLS
+
+typedef enum { MY_CODE_ONE = 0, MY_CODE_TWO = 1 } MyCode;
+
+G_DECLARE_FINAL_TYPE(MyMessageData, my_message_data, MY, MESSAGE_DATA, GObject)
+
+MyMessageData* my_message_data_new(MyCode code, FlValue* data);
```

Comment:
In this particular case, yes since you only receive messages with this object - should we only expose this function if you have a method that has an objects of this type in its args?

Reply:
We would want people to be able to make one of the objects, just not from the `FLValue` serialization. An non-internal construction would be based on the actual ivars like a normal object constructor.

Reply:
Never mind, I was reading this completely wrong. I forgot this object has a field called `data`; this is the constructor for non-nullable fields. I thought `data` was the Pigeon serialization. Ignore this whole thread :)